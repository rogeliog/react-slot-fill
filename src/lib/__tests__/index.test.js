import React from 'react';
import renderer from 'react-test-renderer';
import { Fill, Slot } from '../';

const Toolbar = (props) =>
  <div>
    <Slot name="Toolbar.Item" />
  </div>
    
Toolbar.Item = ({ label }) =>
  <Fill name="Toolbar.Item">
    <button>{label}</button>
  </Fill>

const Footer = (props) =>
  <div>
    <Slot name="Footer.Item" />
  </div>

Footer.Item = ({ href, label }) =>
  <Fill name="Footer.Item">
    <a href={href}>{label}</a>
  </Fill>

it('Fills the a simple slot', () => {
  const Feature = () => <Toolbar.Item label="Home" />;

  const fillComponent = renderer.create([
    <Toolbar />,
    <Feature />,
  ]);

  expect(fillComponent).toMatchSnapshot();
})

it('Fills the appropriate slot', () => {
  const Feature = () => [
    <Toolbar.Item label="Home" />,
    <Toolbar.Item label="About" />,
    <Footer.Item label="Twitter" href="twitter.com/reactjs" />,
  ];

  const fillComponent = renderer.create([
      <Toolbar />,
      <Footer />,
      <Feature />,
    ]);

  expect(fillComponent).toMatchSnapshot();
});