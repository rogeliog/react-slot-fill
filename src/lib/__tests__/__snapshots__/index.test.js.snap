exports[`test Fills the a simple slot 1`] = `
<div>
  <button>
    Home
  </button>
</div>
`;

exports[`test Fills the appropriate slot 1`] = `
Array [
  <div>
    <button>
        Home
    </button>
    <button>
        Home
    </button>
    <button>
        About
    </button>
</div>,
  <div>
    <a
        href="twitter.com/reactjs">
        Twitter
    </a>
</div>,
]
`;
